--- libcxgb3-1.2.5/src/verbs.c.api	2009-09-29 14:57:44.000000000 -0400
+++ libcxgb3-1.2.5/src/verbs.c	2009-11-06 12:37:33.172964811 -0500
@@ -108,7 +108,7 @@ int iwch_free_pd(struct ibv_pd *pd)
 
 static struct ibv_mr *__iwch_reg_mr(struct ibv_pd *pd, void *addr,
 				    size_t length, uint64_t hca_va,
-				    enum ibv_access_flags access)
+				    int access)
 {
 	struct iwch_mr *mhp;
 	struct ibv_reg_mr cmd;
@@ -144,7 +144,7 @@ static struct ibv_mr *__iwch_reg_mr(stru
 }
 
 struct ibv_mr *iwch_reg_mr(struct ibv_pd *pd, void *addr,
-			   size_t length, enum ibv_access_flags access)
+			   size_t length, int access)
 {
 	PDBG("%s addr %p length %ld\n", __FUNCTION__, addr, length);
 	return __iwch_reg_mr(pd, addr, length, (uintptr_t) addr, access);
@@ -267,7 +267,7 @@ struct ibv_srq *iwch_create_srq(struct i
 }
 
 int iwch_modify_srq(struct ibv_srq *srq, struct ibv_srq_attr *attr, 
-		    enum ibv_srq_attr_mask attr_mask)
+		    int attr_mask)
 {
 	return -ENOSYS;
 }
@@ -370,7 +370,7 @@ static void reset_qp(struct iwch_qp *qhp
 }
 
 int iwch_modify_qp(struct ibv_qp *ibqp, struct ibv_qp_attr *attr,
-		   enum ibv_qp_attr_mask attr_mask)
+		   int attr_mask)
 {
 	struct ibv_modify_qp cmd;
 	struct iwch_qp *qhp = to_iwch_qp(ibqp);
@@ -424,7 +424,7 @@ int iwch_destroy_qp(struct ibv_qp *ibqp)
 }
 
 int iwch_query_qp(struct ibv_qp *qp, struct ibv_qp_attr *attr,
-		  enum ibv_qp_attr_mask attr_mask, struct ibv_qp_init_attr *init_attr)
+		  int attr_mask, struct ibv_qp_init_attr *init_attr)
 {
 	return -ENOSYS;
 }
@@ -439,12 +439,12 @@ int iwch_destroy_ah(struct ibv_ah *ah)
 	return -ENOSYS;
 }
 
-int iwch_attach_mcast(struct ibv_qp *qp, union ibv_gid *gid, uint16_t lid)
+int iwch_attach_mcast(struct ibv_qp *qp, const union ibv_gid *gid, uint16_t lid)
 {
 	return -ENOSYS;
 }
 
-int iwch_detach_mcast(struct ibv_qp *qp, union ibv_gid *gid, uint16_t lid)
+int iwch_detach_mcast(struct ibv_qp *qp, const union ibv_gid *gid, uint16_t lid)
 {
 	return -ENOSYS;
 }
--- libcxgb3-1.2.5/src/iwch.h.api	2009-09-29 14:58:25.000000000 -0400
+++ libcxgb3-1.2.5/src/iwch.h	2009-11-06 12:37:19.981089104 -0500
@@ -150,7 +150,7 @@ extern struct ibv_pd *iwch_alloc_pd(stru
 extern int iwch_free_pd(struct ibv_pd *pd);
 
 extern struct ibv_mr *iwch_reg_mr(struct ibv_pd *pd, void *addr,
-				  size_t length, enum ibv_access_flags access);
+				  size_t length, int access);
 extern int iwch_dereg_mr(struct ibv_mr *mr);
 
 struct ibv_cq *iwch_create_cq(struct ibv_context *context, int cqe,
@@ -168,7 +168,7 @@ extern struct ibv_srq *iwch_create_srq(s
 				       struct ibv_srq_init_attr *attr);
 extern int iwch_modify_srq(struct ibv_srq *srq,
 			   struct ibv_srq_attr *attr,
-			   enum ibv_srq_attr_mask mask);
+			   int mask);
 extern int iwch_destroy_srq(struct ibv_srq *srq);
 extern int iwch_post_srq_recv(struct ibv_srq *ibsrq,
 			      struct ibv_recv_wr *wr,
@@ -177,11 +177,11 @@ extern int iwch_post_srq_recv(struct ibv
 extern struct ibv_qp *iwch_create_qp(struct ibv_pd *pd,
 				     struct ibv_qp_init_attr *attr);
 extern int iwch_modify_qp(struct ibv_qp *qp, struct ibv_qp_attr *attr,
-			  enum ibv_qp_attr_mask attr_mask);
+			  int attr_mask);
 extern int iwch_destroy_qp(struct ibv_qp *qp);
 extern int iwch_query_qp(struct ibv_qp *qp,
 			 struct ibv_qp_attr *attr,
-			 enum ibv_qp_attr_mask attr_mask,
+			 int attr_mask,
 			 struct ibv_qp_init_attr *init_attr);
 extern void iwch_flush_qp(struct iwch_qp *qhp);
 extern int t3a_post_send(struct ibv_qp *ibqp, struct ibv_send_wr *wr,
@@ -195,9 +195,9 @@ extern int t3b_post_recv(struct ibv_qp *
 extern struct ibv_ah *iwch_create_ah(struct ibv_pd *pd,
 			     struct ibv_ah_attr *ah_attr);
 extern int iwch_destroy_ah(struct ibv_ah *ah);
-extern int iwch_attach_mcast(struct ibv_qp *qp, union ibv_gid *gid,
+extern int iwch_attach_mcast(struct ibv_qp *qp, const union ibv_gid *gid,
 			     uint16_t lid);
-extern int iwch_detach_mcast(struct ibv_qp *qp, union ibv_gid *gid,
+extern int iwch_detach_mcast(struct ibv_qp *qp, const union ibv_gid *gid,
 			     uint16_t lid);
 extern void t3b_async_event(struct ibv_async_event *event);
 #ifdef DEBUG
